CONTRIBUTING.md

```diff
@@ -0,0 +1,247 @@
+        :subresource_actions:
+            :get:
+            - :name: read
+              :identifier: coffee_show 
+            :post:
+            - :name: delete
+              :identifier: coffee_delete
+```
+The above configuration is very similar to that of a collection, only it defines `:subcollection_actions:` and `:subresource_actions:`. The coffee options will also need to be updated to include the Donuts subcollection:
+```
+    :coffees:
+        :description: Coffee
+        :options: 
+        - :collection
+        :verbs: *gpd
+        :klass: Coffee
+        :subcollections:
+        - :donuts
+```
+
+Subcollection actions are defined in [modules](https://github.com/ManageIQ/manageiq-api/tree/4811ec9fcea2424a585560a761e05d56c53535f8/app/controllers/api/subcollections) and included in the collection controller they are a subcollection of. 
+
+Below is a simple example of what the `Donuts` module would look like:
+```
+module Api
+  module Subcollections
+    module Donuts
+      def donuts_query_resource(object)
+        object.donuts
+      end
+      
+      def donuts_delete_resource(_object, type, id, data)
+        delete_resource(type, id, data)
+      end
+  end
+end
+```
+In the above example, the `object` passed to the `donuts_query_resource` will be the `Coffee` object that the donuts are being queried on (the parent resource to the donuts).
+
+The two method signatures `donuts_query_resource(object)` and `donuts_delete_resource(object, type, id, data)` are the normal naming conventions for subcollection actions, `<subcollection name>_<action>_resource`. Like the collection, you can see the target method being generated [here](https://github.com/ManageIQ/manageiq-api/blob/e9a142a2c4d55d8d0dc18d2e342cdfef251d37bf/app/controllers/api/base_controller/manager.rb#L80-L88).
+
+The `CoffeesController` then includes the `Donut` module as follows:
+```
+module Api
+  class CoffeesController < BaseController
+    include Subcollections::Donuts 
+        
+    def order_resource(type, id, data)
+      coffee = resource_search(id, type, collection_class(type))
+      coffee.order(data)
+    rescue => err
+      raise BadRequestError, "Coffee not ordered - #{err}"
+    end 
+  end 
+end 
+```
+
+This subcollection now gives us the endpoints
+```
+GET /api/coffee/:c_id/donuts
+GET /api/coffee/:c_id/donuts/:s_id
+
+POST /api/coffee/:c_id/donuts/:s_id { "action": "delete" }
+```
+
+And just like that, our API now serves both :coffee: and donuts!
+
+## Creating a PR
+Small PRs are always easier to review, and this example would be broken down into two PRs - the Coffee CRUD PR and a Donuts subcollection PR. Examples in the PR description help reviewers to understand what was added, and also helps the users to get up to speed quickly! 
+Here is a snippet of what we might include as an example for ordering a coffee:
+```
+POST /api/coffee/:id
+{
+    "action": "order",
+    "cream": true,
+    "sugar": false
+}
```

Comment:
I know we support above, would prefer primary form here below and leave above when we add support for POST /api/coffees/:id/order.

```
POST /api/coffees/:id
{
  "action" : "order",
  "resource" : {
    "cream" : true,
    "sugar" : false
  }
}
```